Automattic/abacus
# Experiment Results: update row open state handling

Fixes: https://github.com/Automattic/abacus/issues/731

This PR is addressing two main UX issues:
* Preserve the open state of the last active row when the A/B/n variation or the Analysis Strategy are updated
* Prevent the first metric from [automatically expanding](https://github.com/Automattic/abacus/pull/708) outside of the initial page load (eg: after any of the state updates mentioned above)





## How has this been tested?






- Automated tests that cover added/changed functionality
  * Check if automatic expanding is prevented
  * Check if the opened state is persisted

### Additional manual testing instructions (please specify):
  * For any experiment, expand one of the rows and then change the Analysis Strategy. The last expanded row should be still open
  * For an A/B/n experiment, see the first row automatically expanded. Change Base/Change variations and see how the data updates while the row stays open
    * Close the first row to see many rows on the screen. Change Base/Change variations and the rows should stay closed
